src/main/java/coderank/impl/katzcentrality/KatzCentralityCalculator.java

```diff
@@ -27,16 +29,29 @@ public KatzCentralityCalculator(HashSet<KatzNode> nodes) {
             }
         }
         this.nodes = nodes;
+        this.usedNodes = nodes;
     }
 
     public KatzCentralityCalculator(HashSet<Node<T>> initStorage, HashMap<Node<T>, List<Node<T>>> edges) {
         HashMap<Node<T>, KatzNode> storage = new HashMap<>();
         int index = 0;
         for (Node<T> node : initStorage) {
-            KatzNode currentNode = new KatzNode(index++);
-            storage.put(node, currentNode);
-            revStorage.put(currentNode, node);
-            nodes.add(currentNode);
+            if (node.isUsed()) {
+                KatzNode currentNode = new KatzNode(index++);
+                storage.put(node, currentNode);
+                revStorage.put(currentNode, node);
+                nodes.add(currentNode);
+                usedNodes.add(currentNode);
+            }
+        }
+
+        for (Node<T> node : initStorage) {
+            if (!node.isUsed()) {
+                KatzNode currentNode = new KatzNode(index++);
+                storage.put(node, currentNode);
+                revStorage.put(currentNode, node);
+                nodes.add(currentNode);
+            }
```

Comment:
```
        for (Node<T> node : initStorage) {
              KatzNode currentNode = new KatzNode(index++);
              storage.put(node, currentNode);
              revStorage.put(currentNode, node);
              nodes.add(currentNode);
            if (node.isUsed()) {
                usedNodes.add(currentNode);
            }
        }
```

Reply:
I need used nodes all to be added first into the nodes, because its indexes should be from 0 to (amount of used nodes).
Adjacency matrix have (amount of used nodes) rows and (amount of used nodes) cols and its indexes refer to KatzNode indexes.